src/Components/Homepage/Category-Item/CategoryItem.jsx

```diff
@@ -1,9 +1,20 @@
+import { useNavigate } from "react-router-dom";
+import { useFilter } from "../../../CustomHooks/useFilter";
 import "./CategoryItem.css";
 
 const CategoryItem = ({ title, src }) => {
+
+  const navigate = useNavigate();
+  const { filterDispatch } = useFilter();
+
   return (
-    <div className="category-item">
-      <div
+    <div
+      className="category-item"
+      onClick={() => {
+        filterDispatch({ type: "ADD_CATEGORY", payload: title });
+        navigate("/products");
```

Comment:
In JSX, If writing more than one line of code, then it would be good to move that in a function and call it here. 
So the JSX will look much cleaner.